### ISSUE TITLE
Self-lint false positive

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#200]
#179 currently causes a spurious self-lint error.

```
/home/fengyang/.julia/v0.5/Lint/src/dict.jl:16 E321 lexicalfirst: use of undeclared symbol
/home/fengyang/.julia/v0.5/Lint/src/dict.jl:16 E321 lexicallast: use of undeclared symbol
```

I have decided that it's best to disable the self-lint check in #179 until we can figure out why this is happening and fix it.

**Comment by TotalVerb** [comment]
The list of false positives / real problems has grown.

```
/home/travis/.julia/v0.6/Lint/src/statictype.jl:31 E321 EQ_METHOD_FALSE: use of undeclared symbol

/home/travis/.julia/v0.6/Lint/src/statictype.jl:31 E321 EQ_METHOD_FALSE: use of undeclared symbol

/home/travis/.julia/v0.6/Lint/src/statictype.jl:31 E321 EQ_METHOD_FALSE: use of undeclared symbol

/home/travis/.julia/v0.6/Lint/src/statictype.jl:43 I382 f: argument declared but not used

/home/travis/.julia/v0.6/Lint/src/statictype.jl:43 I382 argtypes: argument declared but not used

/home/travis/.julia/v0.6/Lint/src/variables.jl:179 W545 rhstype: previously used variable has apparent type Type{Any}, but now assigned Type{Type}

/home/travis/.julia/v0.6/Lint/src/variables.jl:190 W542 : comparing apparently incompatible types (#1) LHS:Type{Type} RHS:Type{Any}

/home/travis/.julia/v0.6/Lint/src/variables.jl:190 W542 : comparing apparently incompatible types (#1) LHS:Type{Type} RHS:Type{Any}

/home/travis/.julia/v0.6/Lint/src/variables.jl:171 I382 isConst: argument declared but not used
```

Some of these are real lint errors, not false positives, as indeed the arguments have been declared but not used. We should clean those up or add explicit ignores.
